internal/pkg/service/stream/storage/node/coordinator/filerotation/conditions.go

```diff
@@ -52,18 +57,25 @@ func (c fileRotationConditionResult) String() string {
 		return "size"
 	case timeThreshold:
 		return "time"
+	case sinkThrottled:
+		return "sinkThrottled"
 	default:
 		return "unknown"
 	}
 }
 
-func shouldImport(cfg targetConfig.ImportConfig, now, openedAt, expiration time.Time, stats statistics.Value) fileRotationConditionResult {
+func shouldImport(cfg targetConfig.ImportConfig, now, openedAt, expiration time.Time, stats statistics.Value, jobLimit int) fileRotationConditionResult {
```

Comment:
`jobLimit` is incorrect name for this parameter, should be `runningJobs` probably... 

that said I wanna suggest a different solution, see the comment below